.github/workflows/ci.yml

```diff
@@ -0,0 +1,58 @@
+name: Continuous Integration
+on:
+  push:
+    branches:
+      - master
+
+  pull_request:
+    paths:
+      - .github/workflows/cis.yml
+      - pyparsing/*
+      - setup.py
+      - tox.ini
+
+jobs:
+  tests:
+    name: Unit tests
+    runs-on: ${{ matrix.os || 'ubuntu-latest' }}
+    strategy:
+      matrix:
+        include:
+          - python-version: "3.6"
+            toxenv: py36
+          - python-version: "3.7"
+            toxenv: py37
+          - python-version: "3.8"
+            toxenv: py38
+          - python-version: "3.9"
+            toxenv: py39
+          - python-version: "3.10"
+            toxenv: py310
+          - python-version: "3.10"
+            toxenv: py310
+            os: macos-latest
+          - python-version: "pypy-3.7"
+            toxenv: pypy3
+    env:
+      TOXENV: ${{ matrix.toxenv }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox codecov
```

Comment:
Add dependencies for railroad testing, either:

    pyparsing[railroad]

or

    jinja
    railroad-diagrams